src/parser/variable.js

```diff
@@ -209,17 +182,18 @@ module.exports = {
           }
           break;
         case "[":
+        case "{":
           node = this.node("offsetlookup");
           this.next();
           offset = false;
           if (encapsed) {
             offset = this.read_encaps_var_offset();
```

Comment:
can you check `$obj{$foo]` and `$obj[$bar}` they should fail, but in this implementation they may pass